components/Header.js

```diff
@@ -1,21 +1,92 @@
 import Link from 'next/link'
+import { useState, useRef } from 'react';
+
 import styles from '../styles/Header.module.css';
+import useOnClickOutside from '../hooks/useOnClickOutside.js';
 
 export default function Header({ activePage }) {
+    const [isOpen, setIsOpen] = useState(false);
+    const [isOpenSearch, setIsOpenSearch] = useState(false);
+    const mobileNavMenuRef = useRef(null);
+    const mobileNavSearchRef = useRef(null);
+    useOnClickOutside(mobileNavMenuRef, () => setIsOpen(false));
+    useOnClickOutside(mobileNavSearchRef, () => setIsOpenSearch(false));
+
     return (
-        <header>
-           Header
-           
-           <Link href="/">
-                <a className={`${styles.link} ${activePage === 'Home' ?  styles.activeLink : '' }`}>
-                    Home
-                </a>
-            </Link>
-            <Link href="/products">
-                <a className={`${styles.link} ${activePage === 'Products' ?  styles.activeLink : '' }`}>
-                    Products
-                </a>
-            </Link>
-        </header>
+        <React.Fragment>
+
+            <meta name="viewport" content="width=device-width, initial-scale=1"></meta>
+
+            <nav className={styles.navBar}>
+
+                <button
+                    className={styles.menuIcon}
+                    onClick={() => setIsOpen(!isOpen)}>
+                </button>
+
+                <img src="/HMLogo.svg" alt="logo" className={styles.logo} href="/" />
+
+                <div className={styles.navLinks}>
+                    <Link href="/">
+                        <a className={`${styles.links} ${activePage === 'Home' ? styles.activeLink : ''}`}>
+                            Home
+                        </a>
+                    </Link>
+                    <Link href="/products">
+                        <a className={`${styles.links} ${activePage === 'Products' ? styles.activeLink : ''}`}>
+                            Products
+                        </a>
+                    </Link>
+                    <Link href="/blog">
+                        <a className={`${styles.links} ${activePage === 'Blog' ? styles.activeLink : ''}`}>
+                            Blogs
+                        </a>
+                    </Link>
+                    <Link href="/blog">
+                        <a className={`${styles.links} ${activePage === 'Contact Us' ? styles.activeLink : ''}`}>
+                            Contact Us
+                        </a>
+                    </Link>
+                </div>
+
+                <button className={styles.searchIcon} onClick={() => setIsOpenSearch(!isOpenSearch)}></button>
```

Comment:
Right now the search button isn't closing the search once its open. This is because that useOnClickOutside hook is closing it first (since you click outside it), then this is toggling it back open. Couple ways this could be solved, one way I can think of, is to show a different icon if the search toggle is open. So for example an X icon. That X icon would only setIsOpenSearch to false. That way if its open, and the button is clicked, it will only call set to false.
I think this would work (code is untested, make any changes needed):
```suggestion
                {isOpenSearch && <button className={styles.searchIconClose} onClick={() => setIsOpenSearch(false)}></button>}
                {!isOpenSearch && <button className={styles.searchIcon} onClick={() => setIsOpenSearch(true)}></button>}
```

Reply:
works! Thank you